### ISSUE TITLE
Extension bundle is required for hot reload

### ISSUE THREAD

**Issue by pellebjerkestrand** [Azure/azure-functions-host#9586]
Example at https://github.com/pellebjerkestrand/azure-functions-core-tools-restart-bug

Using JavaScript with programming model 4, changes to app code are not reflected while the server is running.

**Comment by pellebjerkestrand** [comment]
Not a huge problem for very small apps, but larger apps can take a while to start up.

Having a watcher that restarts the server makes the feedback loop of the app I work on the most (111 functions) significantly longer, and will probably mean that we can't ever use the new programming model.

**Comment by ejizba** [comment]
Interesting, thanks for the example - it's working for my apps so that was very helpful. Can you try adjusting your host.json to the following default configuration? I have no idea why, but it seems like the extension bundle is necessary to get this to work.

```json
{
  "version": "2.0",
  "logging": {
    "applicationInsights": {
      "samplingSettings": {
        "isEnabled": true,
        "excludedTypes": "Request"
      }
    }
  },
  "extensionBundle": {
    "id": "Microsoft.Azure.Functions.ExtensionBundle",
    "version": "[4.*, 5.0.0)"
  }
}
```

**Comment by pellebjerkestrand** [comment]
I can confirm that using

```
  "extensionBundle": {
    "id": "Microsoft.Azure.Functions.ExtensionBundle",
    "version": "[4.*, 5.0.0)"
  }
```

results in code changes being picked up while the server is running.

**Comment by ejizba** [comment]
This is probably a host issue, so transferred to that repo. Would be nice if hot reload worked without needing an extension bundle

**Comment by pellebjerkestrand** [comment]
Yeah, that would be great. Even having it documented would be better than not (f.ex. in the upgrade guide to v4).

**Comment by jduysen** [comment]
Does anyone know of a fix for this?  If I am on my vpn or work network, having the extensionBundle object in host.json causes it to crash when I run npm run start, but if I remove extensionBundle object, hot reload doesn't work.  If I connect to my mobile hotspot I can start the project no problem with hot reload.  

Is there maybe a url that I need to have whitelisted or something for this to work correctly on my vpn/company network?
